### PR TITLE
Combat Overhaul Part 1: Accuracy and Evasion, Critical Hits and NPCs

### PR DESCRIPTION
**Pull request by Squird5**
Useroth asked me to do a quick look into critical hits and NPC lethality, which has snowballed into this
Since stabbing eyes and ears was one of the easiest ways to kill someone quickly, I decided to look into reducing the chance of hitting those, resulting in a massively overhauled combat system.

## About The Pull Request

* NEW **Accuracy** system
  * Target zone and combat intents play a major factor in the ability to hit your target
  * Weapon skills are more important
  * Simplemobs also made to use a variant of the Accuracy system, so now they can miss attacks too
  * NEW Facing system prevents hitting a part from the opposite side of the body
  * **Targeting the CHEST Always Hits**
* Reworked Dodge and Parry system
  * Dodge has a new, rebuilt formula
    * Heavy armor values penalize dodging, and full plate dodging is highly disincentivized
  * Parry now built around a more situational modifiers based check, similar to the accuracy system
    * Buffs from skills and stats that put you at an advantage, against debuffs from the stats of the opponent
  * New **No Defense** mode, allowing you to toggle off Dodge and Parry completely
  * Various weapons, and weapon intents, give bonuses or penalties to your defenses
* Touched up Grappling mechanics
  * Opposing rolls rewritten
  * Taking an action that is blocked will now try to resist out of the grab to perform the action
* Reworked Charge knockdown
  * More factors in play, such as vision, direction, NEW Worn Armor Class, the defender being in Combat Mode and if either of you have shields
* Critical Hits reduced and retuned, intended to be lower chance and more character ability oriented than pure luck
  * Future plans for an expanded wounds and critical hits system <sub>Likely reliant on a Medical rewrite</sub>
* NPCs rebalanced around the new systems
  * NPCs and Simplemobs will now target different body parts
  * NPCs and Simplemobs are now affected by Stamina and Fatigue
    * Goblins are fast and weak swarm enemies
    * Skeletons are slow and durable, but with no defensive skill

**Miscellaneous:**
* Bows and Guns given a once or thrice over to try and make them feel better balanced to the rest of the combat system and their roles in it
* Num 7 and Num 9 targeting keybinds expanded to now also target the left eye and skull.
* Someone seems to have deleted the Debug map assets YET AGAIN

## Why It's Good For The Game

The combat is extremely over-tuned, with everyone fighting the same ways and with the same few weapons. Shooting eyes and feet at long ranges, stabbing feet with spears, poking eyes with rapiers. This puts a heavy bias on Center Mass, with smaller, more hard to hit body parts being increasingly difficult, allowing you to _whiff_ attacks against them. Chest targeted melee attacks are still guaranteed.

### CODE REVIEW

**Comment by Gyrandola** [comment]
I really look forward to this. Chest should be what most people target, only the most skilled can afford to go for precision hits.

**Comment by john-pawson** [comment]
Really love this, but isn't 40 dmg for the major wounds a bit too much? It means that bastard swords can't deal gruesome cuts and spears can't deal deep punctures unless you have very high STR (At least 14 for spears), and it ends up making them exclusive to *very* high damage weapons like steel maces and the battle axe. Swords should still be lethal and be able bleed out their enemies faster than a cut from a dagger. Maybe lower it to 30? Agree that 20 is too low though tbh

**Comment by zeratino** [comment]
seconded

**Comment by john-pawson** [comment]
image

File: code/modules/surgery/bodyparts/bodypart_wounds.dm [hunk 1]
```diff
@@ -280,58 +286,64 @@
...
-			necessary_damage = 0.9
+			necessary_damage = 0.85
 		if(prob(used) && (damage_dividend >= necessary_damage))
 			if(dislocation_type)
 				attempted_wounds += dislocation_type
 			attempted_wounds += fracture_type
 	if(bclass in GLOB.artery_bclasses)
-		used = round(damage_dividend * 20 + (dam / 3), 1)
+		used = round((user.STAPER * 0.5) + (dam * 0.3), 1)
+		used += (user?.mind?.get_skill_level(/datum/skill/misc/treatment) * 0.2)
```
Comment (wow50000) [inline code/modules/surgery/bodyparts/bodypart_wounds.dm:342]: not a fan of this, skill should increase arteries not medical.

**Comment by Gyrandola** [comment]
Second test

**Comment by RyszardParys** [comment]
Good lord this is merged again..? Can't wait to get tired after walking through 3 tiles in the bog... Absolutely worthless addition.

**Comment by Koteshkin** [comment]
Does it return insta killing crits from the past? I hope it doesnt.

**Comment by Koteshkin** [comment]
Got old insta killing crits returned? I hope not... Its terrible.

**Comment by RyszardParys** [comment]
I wonder when they will realise we don't actually want it...

**Comment by Gyrandola** [comment]
This is the fourth test run and I have yet to see someone say something positive about this.

Find a way to atomize all these changes or give up on it. People just dont like it at all.

**Pull request closed without merging**